pkg/packaging/fetch.go

```diff
@@ -108,6 +114,81 @@ func FetchBinary(ctx context.Context, localCacheDir, name, binaryName, version s
 	return localBinaryPath, nil
 }
 
-func dlTarPath(name, version, platform string) string {
-	return path.Join("kolide", name, platform, fmt.Sprintf("%s-%s.tar.gz", name, version))
+func dlTarPath(name, channelOrVersion, platform, arch string) (string, error) {
+	// Figure out if we're downloading a specific version or a channel
+	isChannel := channelOrVersion == "stable" || channelOrVersion == "beta" || channelOrVersion == "nightly" || channelOrVersion == "alpha"
+
+	if !isChannel {
+		// We're requesting a version, not a channel, so we already know where the download lives.
+		return dlTarPathFromVersion(name, channelOrVersion, platform, arch), nil
+	}
+
+	version, err := getReleaseVersionFromTufRepo(name, channelOrVersion, platform, arch)
+	if err != nil {
+		return "", fmt.Errorf("could not find release version for channel %s: %w", channelOrVersion, err)
+	}
+
+	return dlTarPathFromVersion(name, version, platform, arch), nil
+}
+
+func dlTarPathFromVersion(name, version, platform, arch string) string {
+	return path.Join("kolide", name, platform, arch, fmt.Sprintf("%s-%s.tar.gz", name, version))
+}
+
+//go:embed assets/tuf/root.json
+var rootJson []byte
```

Comment:
I think this is okay. Though we may want some kind of two step, so that we can maintain a newer TUF repo, and not have to fetch/validate everything.

Reply:
That'd be nice to have for the ee/tuf code too -- it's on my list (eventually)

Reply:
https://github.com/kolide/launcher/issues/1577